modules/dnn/src/layers/cpu_kernels/fast_gemm.cpp

```diff
@@ -0,0 +1,246 @@
+                                         (const char *)B, ldb0, ldb1, beta, (char *)C, ldc, sizeof(float));
+    } else
+#endif
+    {
+        CV_Error(Error::StsNotImplemented, "DNN/fastGemm: instruction set not supported");
+    }
+}
+
+void fastGemm(bool trans_a, bool trans_b,
+              float alpha, const Mat &A, const Mat &B,
+              float beta, Mat &C, FastGemmOpt &opt) {
+    CV_CheckTypeEQ(A.type(), CV_32F, "DNN/fastGemm: only support float32 for now");
+    CV_CheckTypeEQ(A.type(), B.type(), "DNN/fastGemm: A and B should have the same type");
+    CV_CheckTypeEQ(B.type(), C.type(), "DNN/fastGemm: B and C should have the same type");
+
+    const auto shape_a = shape(A);
+    CV_CheckEQ(shape_a.size(), static_cast<size_t>(2), "DNN/fastGemm: A must be 2-dimensional");
+    const auto shape_b = shape(B);
+    CV_CheckEQ(shape_b.size(), static_cast<size_t>(2), "DNN/fastGemm: B must be 2-dimensional");
+    const auto shape_c = shape(C);
+    CV_CheckEQ(shape_c.size(), static_cast<size_t>(2), "DNN/fastGemm: C must be 2-dimensional");
+
+    int ma = shape_a[0], na = shape_a[1];
+    int mb = shape_b[0], nb = shape_b[1];
+
+    int lda0 = na, lda1 = 1, ldb0 = nb, ldb1 = 1, ldc = shape_c[1];
+
+    // printf("trans_a=%d, trans_b=%d, ma=%d, na=%d, mb=%d, nb=%d, alpha=%f, beta=%f, lda0=%d, lda1=%d, ldb0=%d, ldb1=%d\n", static_cast<int>(trans_a), static_cast<int>(trans_b), ma, na, mb, nb, alpha, beta, lda0, lda1, ldb0, ldb1);
+
+    const float *a = A.ptr<const float>();
+    const float *b = B.ptr<const float>();
+    float *c = C.ptr<float>();
+
+    fastGemm(trans_a, trans_b, ma, na, mb, nb,
+             alpha, a, lda0, lda1, b, ldb0, ldb1,
+             beta, c, ldc, opt);
+}
+
+void fastGemmBatched(bool trans_a, bool trans_b,
+                     float alpha, const Mat &A, const Mat &B,
+                     float beta, Mat &C, FastGemmOpt &opt) {
+    CV_CheckTypeEQ(A.type(), B.type(), "DNN/fastGemmBatched: A and B should have the same type");
+    CV_CheckTypeEQ(B.type(), C.type(), "DNN/fastGemmBatched: B and C should have the same type");
+    CV_CheckTypeEQ(A.type(), CV_32F, "DNN/fastGemmBatched: only support float32 for now");
+
+    const auto shape_a = shape(A);
+    size_t dims_A = shape_a.size();
+    CV_CheckGE(dims_A, static_cast<size_t>(2), "DNN/fastGemmBatched: A must be n-dimensional (n >= 2)");
+    const auto shape_b = shape(B);
+    CV_CheckEQ(shape_b.size(), static_cast<size_t>(2), "DNN/fastGemmBatched: B must be 2-dimensional");
+    const auto shape_c = shape(C);
+    size_t dims_C = shape_c.size();
+    CV_CheckGE(dims_C, static_cast<size_t>(2), "DNN/fastGemmBatched: C must be n-dimensional (n >= 2)");
+    // CV_CheckEQ(dims_A, dims_C, "DNN/fastGemmBatched: dims_A must be equal to dims_C");
+    // for (size_t i = 0; i < dims_A - 2; i++) {
+    //     CV_CheckEQ(shape_a[i], shape_c[i], "DNN/fastGemmBatched: dims_A[i] must be equal to dims_C[i]");
+    // }
```

Comment:
Is commented code still required or covered by the checks above?

Reply:
I wrote this because in the beginning I thought if A has extra batch, e.g. 2 x 1 x 1024, then the output Y should carry the batch, e.g. 2 x 1 x 1000. However, the googlenet unit test shows the output should be 2 x 1000 instead. So these lines are commented out.

Reply:
Removed these lines